src/applications/disability-benefits/686c-674/config/address-schema.js

```diff
@@ -463,18 +489,89 @@ export const addressUISchema = (
         'ui:required': (formData, index) => {
           let militaryBasePath = livesOnMilitaryBasePath;
           let countryNamePath = `${path}.countryName`;
+
           if (typeof index === 'number') {
             militaryBasePath = insertArrayIndex(livesOnMilitaryBasePath, index);
             countryNamePath = insertArrayIndex(countryNamePath, index);
           }
           const livesOnMilitaryBase = get(militaryBasePath, formData);
           const countryName = get(countryNamePath, formData);
+
           return (
             (countryName && countryName === USA.value) ||
             (isMilitaryBaseAddress && livesOnMilitaryBase)
           );
         },
         'ui:title': 'Postal Code',
+        'ui:validations': [
+          (errors, zipCode, formData, _schema, _uiSchema, _index) => {
+            // copied scheme from city ui:validations
+            const livesOnMilitaryBaseHash = {
+              base: get(livesOnMilitaryBasePath, formData),
+              address: get(`address[${MILITARY_BASE_PATH}]`, formData),
+              childAddressInfo: get(
+                `childAddressInfo.address[${MILITARY_BASE_PATH}]`,
+                formData,
+              ),
+            };
+            const validationKeys = [
+              'base',
+              'address',
+              'childAddressInfo',
+              'stepChildAddress',
+              'stepChildAddressInfo',
+            ];
+
+            if (window.location.href.includes('review-and-submit')) {
+              livesOnMilitaryBaseHash.stepchildAddress = (
+                formData.stepChildren || []
+              ).some(stepChild =>
+                get(`address[${MILITARY_BASE_PATH}]`, stepChild),
+              );
+              livesOnMilitaryBaseHash.stepChildAddressInfo = (
+                formData.childrenToAdd || []
+              ).some(stepChild =>
+                get(
+                  `childAddressInfo.address[${MILITARY_BASE_PATH}]`,
+                  stepChild,
+                ),
+              );
+            }
+
+            if (
+              isMilitaryBaseAddress &&
+              Object.values(livesOnMilitaryBaseHash).includes(true)
```

Comment:
NP: Could we assign the two .some call results to variables then check them explicitly vs using Object.values? 

This currently checks all values in the hash, so a code update in the future might have some unexpected side effects if the future developer isn't aware this check exists.

Reply:
@DLarson-Oddball the hash also has the default vals of base, address, and childAddressInfo. I figured this check with Object.values is just to see whether we even need to run the logic/validations.

Reply:
I agree, if any additional value is added to the api response that equals `true` it would also cause this to be true.